nomad/resource_acl_auth_method.go

```diff
@@ -0,0 +1,407 @@
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"oidc_client_secret": {
+				Description: "",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+			},
+			"oidc_scopes": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"bound_audiences": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"allowed_redirect_uris": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    true,
+			},
+			"discovery_ca_pem": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"signing_algs": {
+				Description: "",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"claim_mappings": {
+				Description: "",
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"list_claim_mappings": {
+				Description: "",
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceACLAuthMethodCreate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	aclAuthMethod, err := generateNomadACLAuthMethod(d)
+	if err != nil {
+		return err
+	}
+
+	// Create our ACL auth method.
+	log.Print("[DEBUG] Creating ACL Auth Method")
+	aclBindingRuleCreateResp, _, err := client.ACLAuthMethods().Create(aclAuthMethod, nil)
+	if err != nil {
+		return fmt.Errorf("error creating ACL Auth Method: %s", err.Error())
+	}
+	log.Printf("[DEBUG] Created ACL Auth Method %q", aclBindingRuleCreateResp.Name)
+	d.SetId(aclBindingRuleCreateResp.Name)
+
+	return resourceACLAuthMethodRead(d, meta)
+}
+
+func resourceACLAuthMethodDelete(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	authMethodName := d.Id()
+
+	// Delete the ACL auth method.
+	log.Printf("[DEBUG] Deleting ACL Auth Method %q", authMethodName)
+	_, err := client.ACLAuthMethods().Delete(authMethodName, nil)
+	if err != nil {
+		return fmt.Errorf("error deleting ACL Auth Method %q: %s", authMethodName, err.Error())
+	}
+	log.Printf("[DEBUG] Deleted ACL Auth Method %q", authMethodName)
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceACLAuthMethodUpdate(d *schema.ResourceData, meta interface{}) error {
+	providerConfig := meta.(ProviderConfig)
+	client := providerConfig.client
+
+	aclBindingRule, err := generateNomadACLAuthMethod(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %#v", aclBindingRule.Config)
```

Comment:
I think this may leak the sensitive values? I'm not sure if Terraform strips them from logs 🤔

Reply:
Oh yeah, good shout, we will.